packages/react/src/components/Storage/FileUploader/FileUploader.tsx

```diff
@@ -45,6 +64,68 @@ export function FileUploader({
 
   // Previewer Methods
 
+  const progressCallback = (index: number) => {
+    return (progress: { loaded: number; total: number }) => {
+      const percentage = Math.floor((progress.loaded / progress.total) * 100);
+      const status = fileStatusesRef.current[index];
+      fileStatusesRef.current[index] =
+        percentage !== 100
+          ? { ...status, percentage, loading: true }
+          : { ...status, percentage, loading: false, success: true };
+      const addPercentage = [...fileStatusesRef.current];
+      setFileStatuses(addPercentage);
+    };
+  };
+
+  const onDelete = () => {
+    //todo delete
+  };
+
+  const onPause = (index: number): (() => void) => {
+    return function () {
+      fileStatuses[index].uploadTask.pause();
+      const statuses = [...fileStatuses];
+      const status = fileStatuses[index];
+
+      statuses[index] = { ...status, paused: true };
+      setFileStatuses(statuses);
+    };
+  };
+
+  const onResume = (index: number): (() => void) => {
+    return function () {
+      fileStatuses[index].uploadTask.resume();
+      const statuses = [...fileStatuses];
+      const status = fileStatuses[index];
+
+      statuses[index] = { ...status, paused: false };
+      setFileStatuses(statuses);
+    };
+  };
+  const onFileClick = () => {
+    // start upload
+    setLoading(true);
+    const uploadTasksTemp: UploadTask[] = [];
+    for (let i = 0; i < files?.length; i++) {
+      const uploadFileName = getFileName(fileNames?.[i], allFileNames[i]);
+
+      const uploadTask: UploadTask = uploadFile({
+        file: files[i],
+        fileName: uploadFileName,
+        level,
+        progressCallback: progressCallback(i),
+      });
+      uploadTasksTemp.push(uploadTask);
+    }
+    fileStatusesRef.current = uploadTasksTemp.map((status, index) => ({
```

Comment:
I couldn't figure out a good way to do this, so I used a Ref. Maybe there is a non ref way of doing this?